app/src/main/java/io/gnosis/safe/ui/transactions/details/TransactionDetailsFragment.kt

```diff
@@ -291,7 +291,7 @@ class TransactionDetailsFragment : BaseViewBindingFragment<FragmentTransactionDe
                     txStatus.setStatus(
                         txType.titleRes,
                         txType.iconRes,
-                        getStringResForStatus(txDetails.txStatus, needsYourConfirmation),
+                        getStringResForStatus(txDetails.txStatus, canSign && awaitingConfirmations),
```

Comment:
Maybe for added readability you could assign this to `needsYourConfirmation` ? I am assuming that `canSign` checks that you are safe owner that hasn't signed yet and `awaitingConfirmation` just needs the confirmation? Seems like the old flag, just split up.

Reply:
it seems to me that you pushed down the calculation of this flag to this method only, right?

Reply:
Yeah. I was lazy. Looking into it once more.